File: benchmark/java/task7_error.java
import java.util.Optional;

public class Parser {
    public record ParseResult(boolean success, int value, String error) {}

    public static ParseResult parseInt(String s) {
        try {
            return new ParseResult(true, Integer.parseInt(s), null);
        } catch (NumberFormatException e) {
            return new ParseResult(false, 0, e.getMessage());
        }
    }
}

File: benchmark/r/task7_error.r
parse_int <- function(s) {
  result <- tryCatch(
    list(success = TRUE, value = as.integer(s)),
    warning = function(w) list(success = FALSE, error = conditionMessage(w)),
    error = function(e) list(success = FALSE, error = conditionMessage(e))
  )
  result
}
